code/src/nuvla/api/api.py

```diff
@@ -387,14 +390,22 @@ def _cimi_get_uri(self, resource_id=None, resource_type=None):
         return resource_id
 
     def _cimi_request(self, method, uri, params=None, json=None, data=None, headers=None):
-        json_header = {'Accept': 'application/json'}
+        default_headers = {'Accept': APPLICATION_JSON,
+                           'Accept-Encoding': 'gzip'}
         if headers:
-            headers.update(json_header)
+            headers.update(default_headers)
         else:
-            headers = json_header
+            headers = default_headers
         endpoint = '{0}/{1}/{2}'.format(self.endpoint, 'api', uri)
         if self._debug and uri != CLOUD_ENTRY_POINT_ID:
             _request_debug(method, endpoint, params, json, data, headers)
+        if self._compress and json is not None:
+            headers['content-encoding'] = 'gzip'
+            data_to_compress = bytes(jsonlib.dumps(json), 'utf-8')
+            headers['content-type'] = APPLICATION_JSON
+            json = None
+            data = gzip.compress(data_to_compress)
+
         response = self.session.request(method, endpoint,
                                         headers=headers,
                                         allow_redirects=False,
```

Comment:
Here the `default_headers` are not really default because they cannot be overwritten.
Also the current code can modify the headers in the caller.
I would do the following:
```python
def _cimi_request(self, method, uri, params=None, json=None, data=None, headers=None):
    _headers = {'Accept': APPLICATION_JSON,
                'Accept-Encoding': 'gzip'}
    if headers:
        _headers.update(headers)

    ...
```